sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -64,6 +64,16 @@ ProgramManager &ProgramManager::getInstance() {
   return GlobalHandler::instance().getProgramManager();
 }
 
+// This function allows for identifying "service" kernels. A SYCL service kernel
+// is a kernel that has not been defined by the user but is instead generated by
+// the SYCL runtime. These are identified by being in the
+// sycl::detail::__sycl_service_kernel__ namespace and are not given a unique
+// kernel identifier and are not reported as part of kernel bundles.
+static bool isServiceKernel(std::string KernelName) {
+  return KernelName.rfind("_ZTSN2cl4sycl6detail23__sycl_service_kernel__", 0) !=
```

Comment:
Having a contract that each "service" kernel is in `__sycl_service_kernel__` namespace, I believe, that it's enough to just look for a `__sycl_service_kernel__` substring in a given kernel name.

Reply:
Sounds okay to me. It loosens the check, but that is unlikely to matter. I have updated the note to make the motivation for the check slightly clearer.